### PR TITLE
grep includes the i flag

### PR DESCRIPTION
**Pull request by ppsleep**
grep includes the i flag to ignored upper/lower case differences

### CODE REVIEW

File: src/grep.js [hunk 1]
```diff
@@ -17,7 +18,8 @@ common.register('grep', _grep, {
 //@ Available options:
 //@
 //@ + `-v`: Invert `regex_filter` (only print non-matching lines).
-//@ + `-l`: Print only filenames of matching files
+//@ + `-l`: Print only filenames of matching files.
+//@ + `-i`: Ignored upper/lower case differences.
```
Comment (nfischer) [inline src/grep.js]: nit: can you change this sentence to simply "Ignore case."?

File: test/resources/case1 [hunk 2]
```diff
@@ -0,0 +1 @@
+Test3
```
Comment (nfischer) [inline test/resources/case1]: This is OK, but could you move these files under `test/resources/grep/case1*`? That should fix the other failures.

Comment (ppsleep) [reply test/resources/case1]: OK, I moved it, and sorry for my thoughtless actions.

File: src/grep.js [hunk 3]
```diff
@@ -48,6 +50,9 @@ function _grep(options, regex, files) {
     }
 
     var contents = file === '-' ? pipe : fs.readFileSync(file, 'utf8');
+    if (options.ignoreCase) {
+      regex = new RegExp(regex, 'i');
```
[the remark this answers is not in the document]
Comment (ppsleep) [reply src/grep.js]: Oh, it's a low-level error, i fixed it, sorry.

**Comment by nfischer** [comment]
@ppsleep thanks for the great contribution!

**Pull request merged**